src/router/__mocks__/mainRouteTestServer.ts

```diff
@@ -0,0 +1,183 @@
+import express from 'express';
+// eslint-disable-next-line import/no-unresolved
+import { Express } from 'express-serve-static-core';
+import { MultiTenancyOptions, getRequestInformation, InvalidResourceError } from 'fhir-works-on-aws-interface';
+
+import { createMainRoute } from '../routes/createMainRoute';
+import RouteHelper from '../routes/routeHelper';
+import { applicationErrorMapper, httpErrorHandler, unknownErrorHandler } from '../routes/errorHandling';
+
+function provideDecodedToken(mainRouter: express.Router, resourceType: string) {
+    const practitionerTenantsDecoded = {
+        sub: 'fake',
+        name: 'not real',
+        iat: 1516239022,
+        'cognito:groups': [
+            'tenantprefix:915b76f7-8744-4010-bd31-a1e4c0d9fc64',
+            'tenantprefix:125545f5-e7e3-4868-898d-092f1023344b',
+            'tenantprefix:fe470d0a-c7e9-4857-a39c-9a06f68b517b',
+            'practitioner',
+        ],
+    };
+
+    // AuthZ
+    mainRouter.use(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        try {
+            const path = RouteHelper.extractResourceUrl(req.method, req.path);
+            const requestInformation = getRequestInformation(req.method, path);
+
+            if (requestInformation.resourceType !== resourceType) {
+                throw new InvalidResourceError('The request info must contain a resource type');
+            }
+
+            res.locals.userIdentity = practitionerTenantsDecoded;
+            next();
+        } catch (e) {
+            next(e);
+        }
```

Comment:
Can be simpler:

```suggestion
            // remove try and catch

            if (requestInformation.resourceType !== resourceType) {
                next(new InvalidResourceError('The request info must contain a resource type'));
                return;
            }

            res.locals.userIdentity = practitionerTenantsDecoded;
            next();
```

![Screen Shot 2021-08-31 at 16 55 49](https://user-images.githubusercontent.com/34807039/131515637-7082da59-87cb-4acd-907d-73a3e4105ea6.png)